presto-orc/src/main/java/com/facebook/presto/orc/metadata/ExceptionWrappingMetadataReader.java

```diff
@@ -106,7 +108,13 @@ public StripeFooter readStripeFooter(List<OrcType> types, InputStream inputStrea
             return delegate.readBloomFilterIndexes(inputStream);
         }
         catch (IOException | RuntimeException e) {
-            throw new OrcCorruptionException(e, orcDataSourceId, "Invalid bloom filter");
+            throw propagate(e, "Invalid bloom filter");
```

Comment:
`throw newOrcCorruptionException(...)` sounds better to me, what do you think?

Reply:
Agreed, will update

Reply:
Actually, I think `propagate` is right here, since we propagate `PrestoException` directly.